src/Toml/Bi/Map.hs

```diff
@@ -381,6 +383,29 @@ _BoundedInteger = BiMap (Right . toInteger) eitherBounded
          in Left $ ArbitraryError msg
       | otherwise = Right (fromIntegral n)
 
+-- | Helper bimap for 'SumType' and 'Data.Text.Text'.
+_SumTypeText :: (Read a, Show a, Enum a, Bounded a) => TomlBiMap a Text
+_SumTypeText = BiMap
+    { forward  = Right . tShow
+    , backward = toSumType
+    }
+  where
+    toSumType :: forall a. (Read a, Show a, Enum a, Bounded a) => Text -> Either TomlBiMapError a
+    toSumType value
+      | value `elem` options =
+          Right (fromText value)
```

Comment:
I think that this implementation can be slightly efficient and more robust using common programming patterns from Haskell. Currently, you are checking the text value and then parsing it. Let's just get rid of parsing 🙂 

You can create value of type `[(Text, a)]` where the first element of a pair is a shown enum value and the second element is a value itself. So instead of `elem + fromText` you can just do [`lookup`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#v:lookup) and pattern-match on `Maybe`.